tests/trainer/test_supporters.py

```diff
@@ -182,3 +191,73 @@ def test_combined_loader_sequence_max_size_cycle():
         assert len(item) == 2
 
     assert idx == len(combined_loader) - 1
+
+
+class IndexedRandomDataset(RandomDataset):
+
+    def __getitem__(self, index):
+        return {"index": index, "batch": self.data[index]}
+
+
+class TestModel(BoringModel):
```

Comment:
Move this inside the test?

Reply:
It needs to be out to be tested with ddp_spawn. Adding them now.